utils/paginator.py

```diff
@@ -0,0 +1,236 @@
+        self.entries = entries
+        self.max_size = max_size
+        self.show_page_length = show_page_length
+
+        # Checking permissions, will raise CannotPaginate if missing a permission
+        if ctx.guild is not None:
+            self.permissions = ctx.channel.permissions_for(ctx.guild.me)
+        else:
+            self.permissions = ctx.channel.permissions_for(ctx.bot.user)
+
+        self.permissions = ctx.channel.permissions_for(ctx.guild.me if ctx.guild is not None else ctx.bot.user)
+
+        perms = self.permissions
+        mandatory_permissions = [
+            perms.send_messages,
+            perms.embed_links,
+            perms.add_reactions,
+            perms.read_message_history,
+        ]
+
+        for permission in mandatory_permissions:
+            if not permission:
+                raise CannotPaginate(
+                    f"""
+            Bot lacks a mandatory permission to paginate:\n\t
+            Permission: {permission}\n\t
+            Server: {self.guild.name} ({self.guild.id})\n\t
+            Command & Author: {ctx.command.name} ({ctx.author.id}) & {ctx.author.name}
+            """
+                )
+
+        self.page = 0
+        self.embed = discord.Embed()
+
+        # Instanciating a base paginator and adding lines
+        self.paginator = paginator = BasePaginator(max_size=self.max_size, prefix="", suffix="")
+
+        for entry in entries:
+            paginator.add_line(entry)
+
+    async def send(self, timeout: t.Union[int, float] = 300.0, **kwargs: t.Union[str, discord.Emoji],) -> None:
+        """
+    Sends a paginator and processes it according to the reactions passed by the author.
+
+    Parameters
+    -----------
+    timeout: Union[int, float]
+        Timeout for the paginator to exit.
+        If reached, it tells the paginator to exit and clear all the reactions.
+        Defaults to 300.0s (5 minutes)
+    kwargs: Union[str, discord.Emoji]
+        Key-Word arguments to override default reaction emojis
+        Can be used like so:
+
+        `
+        await Paginator().send(timeout=400.0, first_page_emoji='👆', next_page_emoji='✌️')
+        `
+        """
+
+        ctx: Context = self.ctx
+        page = self.page
+        embed = self.embed
+        paginator = self.paginator
+
+        embed.description = paginator.pages[page]
+
+        # Check if show_page_length is True before adding a footer to show page length
+        if self.show_page_length:
+            embed.set_footer(text=f"Page {page + 1}/{len(paginator.pages)}")
+
+        # Reaction emojis. Can be overridden by kwargs.
+        reaction_emojis = {
+            "first_page_emoji": "⏮️",  # First page
+            "prev_page_emoji": "⬅️",  # Previous Page
+            "next_page_emoji": "➡️",  # Next Page
+            "last_page_emoji": "⏭️",  # Last Page
+            "exit_emoji": "<:trashcan:722035312238526505>",  # Stop paginator and delete message (trashcan)
```

Comment:
Add an Emoji to Get the info about the pagination, and an Emoji to go to a valid page chose by me.

Reply:
Okay